Modules/Relationships/src/Relationships.Application/Relationships/Commands/RejectRelationshipReactivation/Handler.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Backbone.BuildingBlocks.Application.Abstractions.Exceptions;
+using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.EventBus;
+using Backbone.BuildingBlocks.Application.Abstractions.Infrastructure.UserContext;
+using Backbone.DevelopmentKit.Identity.ValueObjects;
+using Backbone.Modules.Relationships.Application.Infrastructure.Persistence.Repository;
+using Backbone.Modules.Relationships.Domain.Aggregates.Relationships;
+using Backbone.Modules.Relationships.Domain.DomainEvents.Outgoing;
+using MediatR;
+
+namespace Backbone.Modules.Relationships.Application.Relationships.Commands.RejectRelationshipReactivation;
+public class Handler : IRequestHandler<RejectRelationshipReactivationCommand, RejectRelationshipReactivationResponse>
+{
+    private readonly IRelationshipsRepository _relationshipsRepository;
+    private readonly IEventBus _eventBus;
+    private readonly IdentityAddress _activeIdentity;
+    private readonly DeviceId _activeDevice;
+
+    public Handler(IRelationshipsRepository relationshipsRepository, IUserContext userContext, IEventBus eventBus)
+    {
+        _relationshipsRepository = relationshipsRepository;
+        _eventBus = eventBus;
+        _activeIdentity = userContext.GetAddress();
+        _activeDevice = userContext.GetDeviceId();
+    }
+
+    public async Task<RejectRelationshipReactivationResponse> Handle(RejectRelationshipReactivationCommand request, CancellationToken cancellationToken)
+    {
+        var relationshipId = RelationshipId.Parse(request.RelationshipId);
+
+        var relationship = await _relationshipsRepository.FindRelationship(relationshipId, _activeIdentity, cancellationToken, track: true) ??
+                           throw new NotFoundException(nameof(Relationship));
```

Comment:
`FindRelationships` cannot return null, can it?

Reply:
now when I look at it... `FindRelationship` calls on `FirstWithId` and that method will throw if null, so no, it can't.